File: app/src/main/java/com/pkfcooperparry/template/ui/form/page/PageActivityModule.java
package com.pkfcooperparry.template.ui.form.page;

import com.pkfcooperparry.template.data.DataManager;
import com.pkfcooperparry.template.utils.rx.SchedulerProvider;

import dagger.Module;
import dagger.Provides;

@Module
public class PageActivityModule {

    @Provides
    PageViewModel provideMainViewModel(DataManager dataManager, SchedulerProvider schedulerProvider) {
        return new PageViewModel(dataManager, schedulerProvider);
    }
}

File: app/src/main/java/com/pkfcooperparry/template/ui/form/page/PageActivityModule.kt
package com.pkfcooperparry.template.ui.form.page

import com.pkfcooperparry.template.data.DataManager
import com.pkfcooperparry.template.utils.rx.SchedulerProvider

import dagger.Module
import dagger.Provides

@Module
class PageActivityModule {

    @Provides
    internal fun provideMainViewModel(dataManager: DataManager, schedulerProvider: SchedulerProvider): PageViewModel {
        return PageViewModel(dataManager, schedulerProvider)
    }
}
